### PR TITLE
Addition of Version Command

### PR DESCRIPTION
**Pull request by 6missedcalls**
Versioning for the Speedway CLI
## Changes

- Addition of Updated Splash Screen
- Addition of linking flags to Goreleaser
- Removal of daemon context from root to main
- Addition of commit, date and tag variables to main

### New Features *(required)*
N/A

### Deprecations *(required)*
N/A

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: internal/prompts/prompts.go [hunk 1]
```diff
@@ -123,3 +123,15 @@ func SplashScreen() string {
...
+
+func UpdateSplash() string {
+	s := `
+**
+*	🏎 Sonr Speedway 🏎
+*	You are not running the latest version of Sonr Speedway			
+*	To update please follow the instructions below:
+*	https://www.sonr.dev/docs/tools/speedway/install#installing-speedway-cli
+**
+`
```
Comment (joshLong145) [inline internal/prompts/prompts.go:135]: might be better to use a string builder for this to better control the formatting characters.

File: internal/version/version.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	latestTag := *rel.TagName            // Latest Tag (version)
+	latestBranch := *rel.TargetCommitish // Latest Branch
+	latestPublish := *rel.PublishedAt    // Latest Publish Date for branch
+
+	// Operating System Information for the current machine
+	// Useful for debugging purposes
+	os := runtime.GOOS
+	// Machine Architecture Information for the current machine
+	// Useful for debugging purposes
+	arch := runtime.GOARCH
```
Comment (joshLong145) [inline internal/version/version.go:63]: nit: this information is a bit much for a strict version command. I think we should move this to `speedway info` to get a more granular breakdown of the current build.

File: internal/version/version.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+	return fmt.Sprintf("Version: %s\nLatest Version: %s\nBranch: %s\nOS: %s\nArch: %s\nDev Branch Updated: %s\n", version, latestTag, latestBranch, os, arch, latestPublish), nil
```
Comment (joshLong145) [inline internal/version/version.go:65]: nit: using a string builder here might be easier to control the formatting of this output.

File: cmd/root.go [hunk 4]
```diff
@@ -30,14 +31,15 @@ func bootstrapRootCommand(ctx context.Context, logger *golog.Logger, buildDir em
 	}
 
 	ctx = context.WithValue(ctx, "DAEMON_PORT", 6868)
-
+	ctx = context.WithValue(ctx, "VERSION", "v1.1.2")
```
Comment (joshLong145) [inline cmd/root.go:34]: can we move this an env file? setting this here will make it hard to script.

**Review by joshLong145 — CHANGES_REQUESTED** [review]

File: internal/version/version.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		return false, err
+	}
+	tag := ctx.Value("TAG").(string)
+	latestTag := *rel.TagName
+
+	// if the latest tag and current version are the same, then the cli is up to date
+	if latestTag == tag {
+		return true, nil
+	}
+	return false, nil
```
Comment (mcjcloud) [inline internal/version/version.go:42]: nit: could be consolidated
```
return latestTag == tag, nil
```

**Review by mcjcloud — APPROVED** [review]

**Review by joshLong145 — APPROVED** [review]

File: internal/version/version.go (hunk 2 above)
Comment (mcjcloud) [reply internal/version/version.go]: +1

**Pull request merged**